url97a7a8bbec_PPilger_OOP_2_tgz-pJ8-TestJ8/src/Termin.java

```diff
@@ -0,0 +1,212 @@
+		this.typus = typus;
+		this.ort = ort;
+		this.zeitraum = new Zeitraum(von, bis);
+		this.posten = new Posten(einnahmen, ausgaben, typus.toString(), bis);
+		this.teilnehmer = teilnehmer;
+		this.orig = null;
+	}
+	
+	public Zeitraum getZeitraum() {
+		return zeitraum;
+	}
+	
+	public Posten getPosten() {
+		return posten;
+	}
+
+	public double getAusgaben() {
+		return posten.getAusgaben();
+	}
+
+	public double getEinnahmen() {
+		return posten.getEinnahmen();
+	}
+
+	/**
+	 * @return Teilnehmerliste. Diese darf nicht geaendert werden!
+	 */
+	public List<Mitglied> getTeilnehmer() {
+		return teilnehmer;
+	}
+
+	/**
+	 * Legt eine Kopie des Termins auf den Undo-Stack.
+	 */
+	private void prepareUpdate() {
+		Termin other = new Termin();
+		other.typus = typus;
+
+		// flache kopie (kann nicht geaendert werden, da privat)
+		other.zeitraum = zeitraum;
+
+		// flache kopie (eine aenderung in ort aendert nichts an der bedeutung)
+		other.ort = ort;
+
+		// flache kopie (unveraenderbar)
+		other.posten = posten;
+
+		// flache kopie (aenderungen sind nicht zugelassen)
+		other.teilnehmer = teilnehmer;
+
+		// haenge other hinter this in die historie ein
+		other.orig = orig;
+		this.orig = other;
+	}
+
+	public boolean undo() {
+		if (orig == null) {
+			return false;
+		}
+
+		meldeUpdate("zurueckgesetzt auf vorige Version");
+
+		this.typus = orig.typus;
+		this.ort = orig.ort;
+		this.zeitraum = orig.zeitraum;
+		this.posten = orig.posten;
+		this.teilnehmer = orig.teilnehmer;
+		this.orig = orig.orig;
+
+		return true;
+	}
+
+	/**
+	 * Benachrichtigt alle Teilnehmer ueber die gemachte Aenderung.
+	 * @param aenderung
+	 */
+	private void meldeUpdate(String aenderung) {
+		for (Mitglied t : teilnehmer) {
+			t.sende(orig + " wurde geaendert: " + aenderung);
+		}
+	}
+
+	/**
+	 * @author Christian Kletzander
+	 * @param ort
+	 *            ueberspeichern des Ortes
+	 */
+	public void setOrt(Ort ort) {
+		this.prepareUpdate();
+		this.ort = ort;
+		this.meldeUpdate(orig.ort + " -> " + ort);
+	}
+
+	/**
+	 * @author Christian Kletzander
+	 * @param zeitraum
+	 *            ueberspeichern des Zeitraums
+	 */
+	public void setZeitraum(Date von, Date bis) {
+		this.prepareUpdate();
+		this.zeitraum = new Zeitraum(von, bis);
+		this.meldeUpdate(orig.zeitraum + " -> " + zeitraum);
+	}
+
+	public void setAusgaben(double kosten) {
+		this.prepareUpdate();
+		this.posten = new Posten(posten.getEinnahmen(), kosten,
+				typus.toString(), zeitraum.getLast());
+		this.meldeUpdate("Kosten: " + orig.getAusgaben() + " -> "
+				+ getAusgaben());
+	}
+
+	public void setEinnahmen(double umsatz) {
+		this.prepareUpdate();
+		this.posten = new Posten(umsatz, posten.getAusgaben(),
+				typus.toString(), zeitraum.getLast());
+		this.meldeUpdate("Umsatz: " + orig.getEinnahmen() + " -> "
+				+ getEinnahmen());
+	}
+
+	@Override
+	public String toString() {
+		return typus + ": " + ort + " "
+				+ zeitraum.toString(new SimpleDateFormat("dd.MM.yyyy hh:mm"));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The SimpleDateFormat object was created without setting the time zone. Make sure that you want to use the default time zone or use letter z, Z or X in the pattern. Otherwise, if you do not call setTimeZone() on the created SimpleDateFormat object, you might get an unexpected date and time when using the object.